src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -946,6 +946,178 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
+        /// <param name="requestPath">Endpoint for the REST request</param>
+        /// <param name="expectedErrorMessage">Right error message that should be shown to the end user</param>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(RestMethod.Get, "/api/Book/id/one", null, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request with incorrect primary key parameter type on a table in production mode for MsSql database")]
+        [DataRow(RestMethod.Get, "/api/books_view_all/id/one", null, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request with incorrect primary key parameter type on a view in production mode for MsSql")]
+        [DataRow(RestMethod.Get, "/api/GetBook?id=one", TestHelper.REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request on a stored-procedure with incorrect parameter type in production mode for MsSql")]
+        [DataRow(RestMethod.Post, "/api/Book", TestHelper.REQUEST_BODY_WITH_INCORRECT_PARAM_TYPES, "Invalid value provided for field: publisher_id", DisplayName = "Validates the error message for a POST request with incorrect parameter type in the request body on a table in production mode for MsSql")]
+        [DataRow(RestMethod.Put, "/api/Book/id/one", TestHelper.REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a PUT request with incorrect primary key parameter type on a table in production mode for MsSql")]
+        [DataRow(RestMethod.Put, "/api/Book/id/1", TestHelper.REQUEST_BODY_WITH_INCORRECT_PARAM_TYPES, "Invalid value provided for field: publisher_id", DisplayName = "Validates the error message for a bad PUT request with incorrect parameter type in the request body on a table in production mode for MsSql")]
+        [DataRow(RestMethod.Patch, "/api/Book/id/one", TestHelper.REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a PATCH request with incorrect primary key parameter type on a table in production mode for MsSql")]
+        [DataRow(RestMethod.Patch, "/api/Book/id/1", TestHelper.REQUEST_BODY_WITH_INCORRECT_PARAM_TYPES, "Invalid value provided for field: publisher_id", DisplayName = "Validates the error message for a PATCH request with incorrect parameter type in the request body on a table in production mode for MsSql")]
+        [DataRow(RestMethod.Delete, "/api/Book/id/one", TestHelper.REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a DELETE request with incorrect primary key parameter type on a table in production mode for MsSql")]
+
+        public async Task TestErrorMessageForRestApiForMsSql(
```

Comment:
A method name like `TestGenericExceptionInProductionMode` would be more appropriate. Also if we end up having only one test method rather than having it for every database, we can specify in the method summary that the expected behaviour is same for the other databases as well.

Reply:
While I agree that, at the moment, this code is common across all SQL DB types, having individual tests for each database type helps write tests easier in the future if at all code paths diverge. Right now, when the code path is common, it serves as a sanity (that generic error messages are thrown by all database types) at the very least. My take here would be to not remove test cases.

Reply:
@Aniruddh25 @seantleonard Please share your thoughts

Reply:
If `ParseParamAsSystemType` doesnt hit the db, and we error out in dab itself - adding other db tests right now seems code redundancy to me. 

We have similar situations in this configuration tests where we pick a sample db say MsSql and perform the tests only for MsSql. You may notice MsSql has larger number of tests for this reason.

Reply:
Thanks @ayush3797 and @Aniruddh25 for your inputs. I'll remove the test cases for MySql and PostgreSql